src/parser.ts

```diff
@@ -158,41 +160,54 @@ export function parseLogStream(logStream: QuasiSubject<string>): QuasiObservable
           meta,
           rowIndex: progress.index++,
           row,
+          info: parseJsonInfo(row.Info),
         })
-      });
+      }).catch(reason => console.error(reason));
     }
 
   }, (err) => { console.error('parsing error: ' + err); result.complete(); }, () => {
     if (!progress.completed) {
-      result.complete();
+      setTimeout(() => {
+        result.complete();
+        progress.completed = true;
+      }, 0);
     }
   });
+
   return result;
 }
 
 export function parseLog(log: String): Promise<FlightLog> {
   const lines = log.split('\n');
+  const subject = new QuasiSubject<string>();
+  const parse = parseLogStream(subject);
 
-  if (lines[lines.length - 1] === '') {
-    lines.pop();
+  const flightLog: FlightLog = {
+    metaData: {} as FlightLogMetaData,
+    rows: []
   }
+  parse.subscribe((event) => {
+    // The metadata is updated as the file is parsed, so always grab the latest one.
+    flightLog.metaData = (event.meta) ? event.meta : flightLog.metaData;
+    if (event.row) {
+      flightLog.rows.push(event.row);
+    }
+  });
+  lines.forEach(l => subject.next(l));
+  subject.complete();
 
-  const header = lines.slice(0, LOG_HEADER_LINES);
-  const footer = lines.slice(-LOG_FOOTER_LINES);
-  const body = lines.slice(LOG_HEADER_LINES, -LOG_FOOTER_LINES);
-  const metaData = parseMetaData(header, footer);
-
-  return parseBody(body).then((rows) => ({
-    metaData,
-    rows,
-  }));
+  return new Promise<FlightLog>((resolve, reject) => {
+    parse.toPromise().then(() => resolve(flightLog)).catch((reason => reject(reason)));
+  });
```

Comment:
Can we just simplify to `return parse.toPromise()`? I don't see a reason to wrap it since we aren't doing anything special.

Reply:
The wrapping ensures that the return value remains consistent with the return value of the previous implementation of `parseLog`